### ISSUE TITLE
Finish Login

### ISSUE THREAD

**Issue by jean1880** [wendelltron/tekapp#10]
#### User Story
As a user, I would like to login to the app so that I can create topics, or post on the forum

====
####
- [ ] Set User login identity on succesful login
- [ ] Have option to remember identity for next time the app is opened
- [ ] Have user avatar appear in navbar, to show login status
- [ ] Have option for user to log out

**Comment by jean1880** [comment]
Currently, on android, login is succeeding, and getting the login cookies, and adding them to the app, outstanding issues

- [ ] User variable not being detected properly in navbar
- [ ] Not setting user avatar in navbar
- [ ] Presumably broken for iOS

**Comment by DaAwesomeP** [comment]
Login/logout buttons as well as a generic user icon now show based on `$user.loggedIn`.

**Comment by DaAwesomeP** [comment]
The login token, username, avatar, and logged in status are now stored in localStorage. This should make permanent logins possible.

**Comment by jean1880** [comment]
yes, i can pass the data I was grabbing on login through the service, this is currently on hold while I finish my personal stuff, but will be back on track next week

**Comment by DaAwesomeP** [comment]
The user's profile from the Discourse API is now stored at app launch and login. The user avatar also appears in the navbar.

**Comment by DaAwesomeP** [comment]
It seems that there is no actual login happening. I can type in any user with any password and it will act as if I am logged in as that user.

**Comment by jean1880** [comment]
I'll take a look, but I suspect it is because I am just checking to see if the cookies are set app side. We will have to verify that the cookies are valid on the server, or if we can think of a better solution, I am all ears. The login side is a pain in the ass at the moment, once we have the beta complete, we will need a clever solution for this.

**Comment by DaAwesomeP** [comment]
Also, we are logging into the main site, not Discourse. @wendelltron how do we trigger the page that checks for login on the main site and then logs the user into DIscourse? Is there a better way to do this? I think that it will be messy to login to one sire and then redirect to another without an actual browser UI (headless login).

**Comment by jean1880** [comment]
The login is tracked with the cookie _t, so the only way to login is through the main site, which gives you your token through the cookie
